Lib/ufo2ft/otfPostProcessor.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import print_function, division, absolute_import, unicode_literals
+
+import tempfile
+
+from fontTools.ttLib import TTFont
+
+
+class OTFPostProcessor(object):
+    """Does some post-processing operations on a compiled OpenType font, using
+    info from the source UFO where necessary.
+    """
+
+    def __init__(self, otf, ufo):
+        self.ufo = ufo
+        tmp_file = tempfile.NamedTemporaryFile()
+        tmp_path = tmp_file.name
+        otf.save(tmp_path)
+        self.otf = TTFont(tmp_path)
+
+    def process(self, useProductionNames=True):
+        if useProductionNames:
+            self._rename_glyphs_from_ufo()
+        return self.otf
+
+    def _rename_glyphs_from_ufo(self):
```

Comment:
There's a slight risk there if there's no post table, or post table is format 3.  In that case, the getGlyphOrder() will cause cmap to be loaded, which will then fail to compile after rename.  But if we're sure that the proper post table is there, it's safe to do it this way.